### PR TITLE
Add Storage Health Check Unit Tests

### PR DESCRIPTION
**Pull request by gmfrasca**
## Description of your changes:
Adds unit test coverage for the object storage HealthCheck

## Testing instructions
`make test`.  Alternatively, just make sure the CI checks below pass :)

## Checklist
- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by dsp-developers** [comment]
A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-293`
An OCP cluster where you are logged in as cluster admin is required.

To use this image run the following:

```bash
cd $(mktemp -d)
git clone git@github.com:opendatahub-io/data-science-pipelines-operator.git
cd data-science-pipelines-operator/
git fetch origin pull/293/head
git checkout -b pullrequest 7143de87c24093cb606b1696f01431468e9b394d
make deploy IMG="quay.io/opendatahub/data-science-pipelines-operator:pr-293"
```

More instructions [here](https://github.com/opendatahub-io/data-science-pipelines-operator#deploy-dsp-instance) on how to deploy and test a Data Science Pipelines Application.

File: controllers/storage_test.go [hunk 1]
```diff
@@ -181,3 +185,262 @@ func TestDefaultDeployBehaviorStorage(t *testing.T) {
...
+			},
+		},
+	}
+	dspa.Name = testDSPAName
+	dspa.Namespace = testNamespace
+
+	// Create Context, Fake Controller and Params (unused)
+	ctx, _, reconciler := CreateNewTestObjects()
+
+	secure_connect := false
```
Comment (HumairAK) [inline controllers/storage_test.go:212]: can you camelcase this (same with other occurrences)

File: controllers/storage_test.go [hunk 2]
```diff
@@ -181,3 +185,262 @@ func TestDefaultDeployBehaviorStorage(t *testing.T) {
...
+			Host:            "foo",
+			Port:            "1337",
+			Secure:          &SecureConnection,
+			AccessKeyID:     base64.StdEncoding.EncodeToString([]byte("fooaccesskey")),
+			SecretAccessKey: base64.StdEncoding.EncodeToString([]byte("foosecretkey")),
+		},
+	}
+
+	verified := reconciler.isObjectStorageAccessible(ctx, dspa, params)
+	assert.True(t, verified) // if health check is disabled this should always return True even thought the mock connection function would return false if called
```
Comment (HumairAK) [inline controllers/storage_test.go:300]: can we multi line this comment and not make it inline

File: controllers/storage_test.go [hunk 3]
```diff
@@ -181,3 +185,251 @@ func TestDefaultDeployBehaviorStorage(t *testing.T) {
...
+func TestJoinHostPort(t *testing.T) {
+	tests := map[string]struct {
+		host     string
+		port     string
+		expected string
+	}{
+		"host and port defined": {host: "somehost", port: "1234", expected: "somehost:1234"},
+		"empty port":            {host: "somehost", port: "", expected: "somehost"},
+		"empty host":            {host: "", port: "1234", expected: ":1234"}, // TODO: this obviously wouldn't actually work, should we handle this case?
+		"both empty":            {host: "", port: "", expected: ""},          // TODO: this obviously wouldn't actually work, should we handle this case?
```
Comment (HumairAK) [inline controllers/storage_test.go:390]: I think we should error if host is not provided,

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-293`

File: controllers/storage_test.go (hunk 3 above)
Comment (gmfrasca) [reply controllers/storage_test.go]: added a new commit to handle it, removed these TODO comments

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-293`

**Pull request merged**